tests/MongoDB.Bson.Tests/ObjectModel/BsonObjectIdTests.cs

```diff
@@ -28,7 +27,7 @@ public void TestByteArrayConstructor()
 #pragma warning disable 618
             byte[] bytes = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12 };
             var objectId = new BsonObjectId(bytes);
-            Assert.Equal(0x01020304, objectId.Timestamp);
+            Assert.Equal((uint)0x01020304, objectId.Timestamp);
```

Comment:
This seems to demonstrate that the change is not only binary breaking but also source breaking.  Is that assessment correct?

Reply:
By default, 0x##### values are considered as `int` typed. So this changes is required, since `Assert.Equals` doesn't contain overloads that expect `int` and `uint` income parameters.
However, this code will work:

    (objectId.Timestamp == 0x01020304).Should().BeTrue();

But I agree, that this change looks like a source breaking change.

>Is that assessment correct?

I think so

Reply:
To avoid breaking changes, I can leave the constructors and methods from the previous implementation (which expects `int` except `uint`) with marking them as `obsolete`, but I cannot do it for this line (without property renaming): https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/85/files#diff-312b5ea5d26aab247a43a999283d4b53R139. Thoughts?
cc: @rstam, @jyemin

Reply:
If you accept my suggestions in `ObjectId.cs` this file needs no changes at all.